mongodbatlas/resource_mongodbatlas_organization.go

```diff
@@ -0,0 +1,179 @@
+		},
+		Schema: map[string]*schema.Schema{
+			"org_owner_id": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"description": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"org_id": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"public_key": {
+				Type:      schema.TypeString,
+				Computed:  true,
+				Sensitive: true,
+			},
+			"private_key": {
+				Type:      schema.TypeString,
+				Computed:  true,
+				Sensitive: true,
+			},
+			"role_names": {
+				Type:     schema.TypeSet,
+				Required: true,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+			},
+		},
+	}
+}
+
+func resourceMongoDBAtlasOrganizationCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	conn := meta.(*MongoDBClient).Atlas
+
+	createRequest := new(matlas.CreateOrganizationRequest)
+	apiKey := &matlas.APIKeyInput{}
+
+	createRequest.Name = d.Get("name").(string)
+	createRequest.OrgOwnerID = pointy.String(d.Get("org_owner_id").(string))
+	apiKey.Desc = d.Get("description").(string)
+	apiKey.Roles = expandStringList(d.Get("role_names").(*schema.Set).List())
+
+	organization, resp, err := conn.Organizations.Create(ctx, &matlas.CreateOrganizationRequest{
+		Name:       createRequest.Name,
+		OrgOwnerID: createRequest.OrgOwnerID,
+		APIKey:     apiKey,
+	})
+	if err != nil {
+		if resp != nil && resp.StatusCode == http.StatusNotFound {
+			d.SetId("")
+			return nil
+		}
+
+		return diag.FromErr(fmt.Errorf("error create Organization: %s", err))
+	}
+
+	if err := d.Set("private_key", organization.APIKey.PrivateKey); err != nil {
+		return diag.FromErr(fmt.Errorf("error setting `public_key`: %s", err))
+	}
+
+	if err := d.Set("public_key", organization.APIKey.PublicKey); err != nil {
+		return diag.FromErr(fmt.Errorf("error setting `public_key`: %s", err))
+	}
+
+	if err := d.Set("org_id", organization.Organization.ID); err != nil {
+		return diag.FromErr(fmt.Errorf("error setting `org_id`: %s", err))
+	}
+
+	d.SetId(encodeStateID(map[string]string{
+		"org_id": organization.Organization.ID,
+	}))
+
+	return resourceMongoDBAtlasOrganizationRead(ctx, d, meta)
+}
+
+func resourceMongoDBAtlasOrganizationRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	// Get client connection.
+	config := Config{
+		PublicKey:  d.Get("public_key").(string),
+		PrivateKey: d.Get("private_key").(string),
+	}
+
+	clients, _ := config.NewClient(ctx)
+	conn := clients.(*MongoDBClient).Atlas
+
+	ids := decodeStateID(d.Id())
+	orgID := ids["org_id"]
+
+	organization, resp, err := conn.Organizations.Get(ctx, orgID)
+	if err != nil {
+		if resp != nil && resp.StatusCode == http.StatusNotFound || resp.StatusCode == http.StatusBadRequest {
+			log.Printf("warning Organization deleted will recreate: %s \n", err.Error())
```

Comment:
When do you get `StatusBadRequest` instead of `StatusNotFound`?  if the org is deleted, I would expect to get a NotFound

Reply:
Corrected